openfl/utilities/logs.py

```diff
@@ -57,3 +58,51 @@ def setup_loggers(log_level=logging.INFO):
     formatter = logging.Formatter("[%(asctime)s][%(name)s][%(levelname)s] - %(message)s")
     handler.setFormatter(formatter)
     root.addHandler(handler)
+
+
+def get_memory_usage(logger, round_number, metric_origin):
```

Comment:
Instead of passing `round_number` and `metric_origin`, pass a single argument called `msg`, which is a string that gets prefixed on every print statement made by the function. The goal of functions is to have reusability and keep them generic without compromising readability.

Also suggest renaming the function to `log_memory_usage(msg: str)`.

This is my recommended signature:

```python
def log_memory_usage(msg: str = "") -> dict:
  """Logs memory usage details of the current process.
  
  Args:
    msg: Optional prefix message for logging.

  Returns:
    A dict of memory usage details.
  """
  pass
```